internal/pkg/print/debug.go

```diff
@@ -162,25 +177,110 @@ func BuildDebugStrFromHTTPResponse(resp *http.Response, includeHeaders []string)
 	headersMap := buildHeaderMap(resp.Header, includeHeaders)
 	headers := fmt.Sprintf("response headers: %v", BuildDebugStrFromMap(headersMap))
 
-	if resp.Body == nil {
+	bodyMap, err := dumpRespBody(resp)
+	if err != nil {
+		return []string{status, headers}, fmt.Errorf("read response body: %w", err)
+	}
+	if bodyMap == nil {
 		return []string{status, headers}, nil
 	}
-	body, err := io.ReadAll(resp.Body)
+	body := fmt.Sprintf("response body: %s", BuildDebugStrFromMap(bodyMap))
+
+	return []string{status, headers, body}, nil
+}
+
+// dumpRespBody reads the response body and returns a string representation of it.
+// Based on code from httputil package
+// https://pkg.go.dev/net/http/httputil#DumpResponse
+func dumpRespBody(resp *http.Response) (map[string]any, error) {
```

Comment:
Why not use the `DumpRequest` and `DumpResponse` methods directly from the standard library `httputils` pkg? 
If needed, then make a wrapper around those, but I don't see why we need to copy the code here and maintain it ourselves

Reply:
The DumpResponse/Request methods return the entire Response/Request in a format that is not consistent with the other debug logs from the CLI. They also include a lot of formatting characters (\n,\r,...) that don't look good in slog and make it practically impossible to filter the noise (additional headers, etc).
The other problem is that their auxiliary functions (drainBody), which would be particularly useful here, are private.

Reply:
as discussed offline, let's use the initial implementation with the `ReadAll` and `drainBody`

Reply:
No need to have `DumpRequest` and `DumpResponse` methods copied here